lib/src/test/java/io/ably/lib/test/rest/RestAuthAttributeTest.java

```diff
@@ -0,0 +1,95 @@
+package io.ably.lib.test.rest;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.test.common.Setup;
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Capability;
+import io.ably.lib.types.ClientOptions;
+
+import static io.ably.lib.rest.Auth.AuthOptions;
+import static io.ably.lib.rest.Auth.TokenCallback;
+import static io.ably.lib.rest.Auth.TokenDetails;
+import static io.ably.lib.rest.Auth.TokenParams;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/**
+ * Created by VOstopolets on 9/3/16.
+ */
+public class RestAuthAttributeTest {
+
+    private static AblyRest ably;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        Setup.TestVars testVars = Setup.getTestVars();
+        ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+        opts.clientId = "defaultClientId";
+        ably = new AblyRest(opts);
+    }
+
+    /**
+     * Stores the AuthOptions and TokenParams arguments as defaults
+     * for subsequent authorisations with the exception of the attributes:
+     * {@link TokenParams#timestamp}, {@link AuthOptions#queryTime}, {@link AuthOptions#force}
+     * <p>
+     * Spec: RSA10g
+     * </p>
+     */
+    @Test
+    public void auth_stores_options_params() {
+        try {
+            /* init custom TokenParams */
+            final Capability capability1 = new Capability();
+            capability1.addResource("testchannel", "subscribe");
+            TokenParams tokenParams = new TokenParams() {{
+                ttl = 4000L;
+                clientId = "firstClientId";
+                capability = capability1.toString();
+            }};
+
+            /* init custom AuthOptions */
+            final Setup.TestVars testVars = Setup.getTestVars();
+            AuthOptions authOptions = new AuthOptions() {{
+                authCallback = new TokenCallback() {
+                    private AblyRest ably = new AblyRest(testVars.createOptions(testVars.keys[0].keyStr));
+
+                    @Override
+                    public Object getTokenRequest(TokenParams params) throws AblyException {
+                        return ably.auth.requestToken(params, null);
+                    }
+                };
+                key = testVars.keys[1].keyStr;
+            }};
+
+            /* authorise with custom options */
+			TokenDetails tokenDetails1 = ably.auth.authorise(authOptions, tokenParams);
```

Comment:
Sorry, it's old test, and I haven't changed these spaces. I changed my IDE settings, and in the future I hope it will not happen.